src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,27 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public abstract class Contact {
+
+	public Contact(boolean isPrivate) throws IllegalValueException {
+		this.isPrivate = isPrivate;
```

Comment:
Try to look for way to do validation here. The trick is to ensure the subclasses provide their own validation .